olp/utils.py

```diff
@@ -83,6 +83,29 @@ def remove_perm(user, permission, obj=None):
     return True
 
 
+def remove_permission_for_object(model_instance):
+    """
+    Removes a permission for a given object.
+    """
+
+    from django.contrib.contenttypes.models import ContentType
+    from olp.models import ObjectPermission
+
+    # Get the content type for the given instance
+    object_content_type = ContentType.objects.get_for_model(
+        model_instance.__class__
+    )
+
+    # Unlink any permissions associated with this model
+    permissions = ObjectPermission.objects.filter(
+        base_object_id=model_instance.id,
+        base_object_ct=object_content_type
```

Comment:
We should also be filtering for any permissions where the instance is the `target_object`.

```
(base_id = model.id AND base_ct = model_ct) OR (target_id = model.id AND target_ct = model_ct)
```